org.OpenT2T.Sample.SuperPopular.Lamp/Vera Dimmable Light Bulb/js/thingTranslator.js

```diff
@@ -0,0 +1,174 @@
+    relaySessionToken : null,
+
+    // serial number of the vera hub
+    pkDevice : null,
+
+    // data structure which represents the device targetted by this translator
+    device: null,
+
+    initDevice: function(dev) {
+        var deferred = q.defer();
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.relay_session_token) !== 'undefined') {
+                    this.relaySessionToken = props.relay_session_token;
+                } else {
+                    console.log('props.relay_session_token is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                if (typeof (props.relay_server) !== 'undefined') {
+                    this.relayServer = props.relay_server;
+                } else {
+                    console.log('props.relay_server is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                if (typeof (props.pk_device) !== 'undefined') {
+                    this.pkDevice = props.pk_device;
+                } else {
+                    console.log('props.pk_device is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    this.deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                deferred.resolve();
+            } else {
+                console.log('props is undefined.');
+                setImmediate(deferred.reject);
+            }
+        } else {
+            console.log('device is undefined.');
+            setImmediate(deferred.reject);
+        }
+
+        console.log('Javascript initialized.');
+        logDeviceState(this.device);
+
+        return deferred.promise;
+    },
+
+    turnOn: function() {
+        console.log('[' + this.deviceId + '] turnOn called.');
+        return sendPowerStateCommandToDevice(100, this.deviceId, this.relayServer, this.relaySessionToken, this.pkDevice);
+    },
+
+    turnOff: function() {
+        console.log('[' + this.deviceId + '] turnOff called.');
+        return sendPowerStateCommandToDevice(0, this.deviceId, this.relayServer, this.relaySessionToken, this.pkDevice);
+    },
+
+    // sets the dimmable bulb to the desired brightness, valid values: integer 0-100
+    setBrightness: function(brightness) {
+        console.log('[' + this.deviceId + '] setBrightness called with value: ' + brightness);
+        return sendPowerStateCommandToDevice(brightness, this.deviceId, this.relayServer, this.relaySessionToken, this.pkDevice);
+    },
+
+    disconnect: function() {
+        console.log('disconnect called.');
+        var deferred = q.defer();
```

Comment:
Again, no need for a promise here. (For all the methods, the caller will detect whether or not a promise was returned.)